src/dbnode/storage/index.go

```diff
@@ -1226,12 +1229,24 @@ func (i *nsIndex) query(
 
 	// If require exhaustive but not, return error.
 	if opts.RequireExhaustive {
+		seriesCount := results.Size()
+		if opts.SeriesLimitExceeded(seriesCount) {
+			i.metrics.queryNonExhaustiveSeriesLimitError.Inc(1)
+		}
+		docsCount := results.TotalDocsCount()
+		if opts.DocsLimitExceeded(docsCount) {
+			i.metrics.queryNonExhaustiveDocsLimitError.Inc(1)
+		}
 		i.metrics.queryNonExhaustiveLimitError.Inc(1)
```

Comment:
Seems like this metric is not needed anymore? It'll double count if we `sum(rate(query{exhaustive="false"}))` since it will have multiple metrics emitted (one for concrete type, and one for the general case).  Maybe we remove this metric and just rely on counting the concrete error types (series limit error and docs limit error)?

Reply:
Yeah I left around just because *technically* you could have some path that returns `Exhaustive=false` but neither series nor docs limit is exceeded (atm that isn't possible, but I was just being defensive). But agreed it will be confusing to keep this around because of double counting on standard queries, so maybe we do this to avoid the double counting but still capture any cases where those limits are not met:
```
if opts.SeriesLimitExceeded(seriesCount) {
			i.metrics.queryNonExhaustiveSeriesLimitError.Inc(1)
		} else if opts.DocsLimitExceeded(docsCount) {
			i.metrics.queryNonExhaustiveDocsLimitError.Inc(1)
		} else {
			i.metrics.queryNonExhaustiveLimitError.Inc(1)
		}
```
If though we think that is overkill and we should just assert one or the other limits should be met, then we can just remove the generic limit metric.